src/Middleware/RetryMiddleware.php

```diff
@@ -51,14 +51,23 @@ class RetryMiddleware
     /** @var float|null */
     private $deadlineMs;
 
+    /*
+     * The number of retries that have already been attempted.
+     * The first retry will have $retryAttempts set to 0.
+     * @var int
+     */
+    private $retryAttempts;
```

Comment:
We can use typehints! I have a [PR for the rest of the library](https://github.com/googleapis/gax-php/pull/452), but as we add new ones I think it's good to add them: 
```suggestion
    private int $retryAttempts;
```